src/Tooltip.js

```diff
@@ -0,0 +1,174 @@
+
+import { Popper } from './Popper.js'
+import { colors, layers } from './theme.js'
+import { referencePlacementPropType } from './common-prop-types.js'
+;('') // TODO: https://github.com/jsdoc/jsdoc/issues/1718
+
+const TOOLTIP_OFFSET = 4
+
+const popperStyle = resolve`
+    z-index: ${layers.applicationTop};
+    pointer-events: none;
+`
+
+const offsetModifier = {
+    name: 'offset',
+    options: {
+        offset: [0, TOOLTIP_OFFSET],
+    },
+}
+
+const OPEN_DELAY = 200
+const CLOSE_DELAY = 400
+
+/**
+ * @module
+ * @param {Tooltip.PropTypes} props
+ * @returns {React.Component}
+ *
+ * @example import { Tooltip } from '@dhis2/ui-core'
+ *
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/tooltip.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/tooltips--default-fluid|Storybook}
+ */
+class Tooltip extends Component {
+    state = { open: false }
+    ref = createRef()
+    openTimeout = null
+    closeTimeout = null
+
+    componentWillUnmount() {
+        clearTimeout(this.openTimeout)
+        clearTimeout(this.closeTimeout)
+    }
+
+    onMouseOver = () => {
+        clearTimeout(this.closeTimeout)
+
+        this.openTimeout = setTimeout(() => {
+            this.setState({ open: true })
+        }, OPEN_DELAY)
+    }
+
+    onMouseOut = () => {
+        clearTimeout(this.openTimeout)
+
+        this.closeTimeout = setTimeout(() => {
+            this.setState({ open: false })
+        }, CLOSE_DELAY)
+    }
+
+    render() {
+        const {
+            children,
+            className,
+            content,
+            dataTest,
+            display,
+            maxWidth,
+            placement,
+        } = this.props
+        const { open } = this.state
+
+        return (
+            <>
+                <span
+                    onMouseOver={this.onMouseOver}
+                    onMouseOut={this.onMouseOut}
+                    ref={this.ref}
+                    data-test={`${dataTest}-reference`}
+                >
+                    {children}
+                </span>
+
+                {open &&
+                    createPortal(
+                        <Popper
+                            className={popperStyle.className}
+                            placement={placement}
+                            reference={this.ref}
+                            modifiers={[offsetModifier]}
+                        >
+                            <div
+                                className={className}
+                                data-test={`${dataTest}-content`}
+                            >
+                                {content}
+                            </div>
+                        </Popper>,
+                        document.body
+                    )}
+                {popperStyle.styles}
+                <style jsx>{`
+                    span {
+                        display: ${display};
+                    }
```

Comment:
I get the intention, not sure I agree with the solution.
Instead, we could expose an `element` prop. 

Span is a generic element for inline content, while div is the generic element for block content, so using a `span` with `display: block` is kind of wrong (technically it's correct though). Using an `element` prop could add more meaning though and would allow to automatically use styles defined by the app (i. e. when the app defined styles for paragraphs and the anchor is a `p` as well with `element="p"`)

Reply:
The massive list of allowed values for this prop would also disappear

Reply:
Using `element` or `component` would allow custom components as well as long as they support the events passed to the component.

Reply:
I wasn't super stoked about my current solution either, but I see some limitations to both of the options you are suggesting.

**An `element` prop**
This basically only lets us differentiate between `inline` and `block`. Other flavours of display would be unsupported, because this cannot be linked to a specific element. I think I'd prefer to just keep the current `display` prop and just limit it to `inline` and `block` for now, and add allowed values when the need presents itself.

**Passing an `element` / `component`**
I considered this option, but it has a major downside:
- If we'd want to accept this, we wouldn't want to wrap the passed `element` / `component`,  but just render it directly (if we'd keep wrapping that component the situation would be identical to what we have now)
- But the popper needs a ref to its reference/anchor element. So that means that every component that is being passed to the Tooltip needs to implement a `ref`. We can't really enforce this, so it'd be up to the docs...

Reply:
> If we'd want to accept this, we wouldn't want to wrap the passed element / component, but just render it directly (if we'd keep wrapping that component the situation would be identical to what we have now)

We can do it this way:

```jsx
const AComponent = ({ component: Component }) => (
  <Component>
  </Component>
)
```

> But the popper needs a ref to its reference/anchor element. So that means that every component that is being passed to the Tooltip needs to implement a ref. We can't really enforce this, so it'd be up to the docs...

That's true, which just means that a custom component would need to use `forwardRef`, which is available, don't see the issue here? We can't enforce certain props either (we've tried that and agreed that it's impossible), so we already have this "pattern" of "do it right and read the docs or it won't work"

Reply:
OK... I've implemented this in https://github.com/dhis2/ui-core/pull/835/commits/90eba6a331a89fe7ce0591e87d765ea5dd0147af